### PR TITLE
add customizable controller rate limiter

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve?

The current workqueue ratelimiter for controllers are using the default combination of `ItemExponentialFailureRateLimiter` and `BucketRateLimiter`. However the `ItemExponentialFailureRateLimiter` has a `baseDelay` of 5ms and `maxDelay` of 1000s, which is unrealistic.

### What is changed and how does it work?

add a new controller rate limiter function, and default to `baseDelay=1s` and `maxDelay=100s`

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [x] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Optimize CR syncing intervals
```

### CODE REVIEW

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Comment by csuzhangxc** [comment]
@dragonly you need to fix CI....

File: pkg/controller/backupschedule/backup_schedule_controller.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func NewController(deps *controller.Dependencies) *Controller {
 		deps:    deps,
 		control: NewDefaultBackupScheduleControl(controller.NewRealBackupScheduleStatusUpdater(deps), backupschedule.NewBackupScheduleManager(deps)),
 		queue: workqueue.NewNamedRateLimitingQueue(
-			workqueue.DefaultControllerRateLimiter(),
+			controller.NewControllerRateLimiter(1*time.Second, 100*time.Second),
```
Comment (DanielZhangQD) [inline pkg/controller/backupschedule/backup_schedule_controller.go:48]: Please also update the code in pkg/controller/backup/backup_controller.go.

Comment (dragonly) [reply pkg/controller/backupschedule/backup_schedule_controller.go:48]: done

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**